tests/metrics/test_categorical_risk_observer.py

```diff
@@ -0,0 +1,137 @@
+import numpy as np
+import pandas as pd
+import pytest
+from vivarium import InteractiveContext
+from vivarium.testing_utilities import TestPopulation, build_table
+
+from vivarium_public_health.metrics.risk import CategoricalRiskObserver
+from vivarium_public_health.metrics.stratification import (
+    ResultsStratifier as ResultsStratifier_,
+)
+from vivarium_public_health.risks.base_risk import Risk
+from vivarium_public_health.utilities import to_years
+
+
+@pytest.fixture
+def categorical_risk():
+    year_start = 1990
+    year_end = 2010
+    risk = "test_risk"
+    risk_data = dict()
+    exposure_data = build_table(
+        0.25, year_start, year_end, ["age", "year", "sex", "cat1", "cat2", "cat3", "cat4"]
+    ).melt(
+        id_vars=("age_start", "age_end", "year_start", "year_end", "sex"),
+        var_name="parameter",
+        value_name="value",
+    )
+
+    risk_data["exposure"] = exposure_data
+    risk_data["categories"] = {
+        "cat1": "severe",
+        "cat2": "moderate",
+        "cat3": "mild",
+        "cat4": "unexposed",
+    }
+    risk_data["distribution"] = "ordered_polytomous"
+    return Risk(f"risk_factor.{risk}"), risk_data
+
+
+# Subclass of ResultsStratifier for integration testing
+class ResultsStratifier(ResultsStratifier_):
+    configuration_defaults = {
+        "stratification": {
+            "default": ["age_group", "sex"],
+        }
+    }
+
+
+def test_observation_registration(base_config, base_plugins, categorical_risk):
```

Comment:
are base_plugins and categorical_risk defined in a conf.py?

Reply:
base_config and base_plugins (which I think is what you meant) are in tests/conftest.py and categorical_risk is in this file.